### PR TITLE
Add `states()` method to return actual genotype states as characters

### PR DESCRIPTION
**Pull request by hyanwong**
## Description

Adds `variant.genotype_values()`. See https://github.com/tskit-dev/tsinfer/issues/739

# PR Checklist:

- [x] Tests that fully cover new/changed functionality.
- [x] Documentation including tutorial content if appropriate.
- [x] Changelogs, if there are API changes.

### CODE REVIEW

File: python/tskit/genotypes.py
Comment (benjeffery) [inline python/tskit/genotypes.py:253]: Can we render this in a warning box?

Comment (hyanwong) [reply python/tskit/genotypes.py:253]: Yes, good idea

Comment (hyanwong) [reply python/tskit/genotypes.py]: Done. Is the method name OK, do you think?

**Comment by hyanwong** [comment]
I'm not sure why this is failing CI: everything is covered AFAICS.

Ready to merge, I reckon, pending approval of the name of the function.

**Review by jeromekelleher — COMMENTED** [review]
LGTM - not sure about the name though.

**Comment by jeromekelleher** [comment]
Also not clear that an Object array is what people would want. Would it be better to have a "missing data symbol" or something like we do for ``haplotypes``?

**Comment by hyanwong** [comment]
> Also not clear that an Object array is what people would want. Would it be better to have a "missing data symbol" or something like we do for `haplotypes`?

ISWYM, but using any particular character will mean that we can't use that for an allele type (which we might conceivably want to do). This is only going to bite for missing data, so I don't think it's too bad TBH.

File: python/tskit/genotypes.py
Comment (hyanwong) [reply python/tskit/genotypes.py:248]: SGTM

**Comment by jeromekelleher** [comment]
Hmm - how about writing some client code for this with and without missing data? Something like the ``verify`` function in tsinfer would be a good one. We definitely don't want to have to have different code paths for dtypes with and without missing data.

We use "N" as the default missing data character elsewhere and its fine. We should be able to use this function to compare easily with e.g. ``alignments``. 

Is the ``np.array(list(ts.alignments())`` equal to ``np.array(var.states() for var in ts.variants()).T`` with and without missing data?

File: python/tskit/genotypes.py
Comment (hyanwong) [reply python/tskit/genotypes.py:248]: you mean `v.genotypes_state()` or simply `v.states()`

**Comment by benjeffery** [comment]
@hyanwong Would you like this to be in 0.5.4?

**Comment by hyanwong** [comment]
Ah, good point. There's no hurry from my PoV. Is it just a question of my changing the name? I can do that easily enough.

**Comment by jeromekelleher** [comment]
I don't think we should hurry this in  as we don't want to get stuck with Object arrays unless we have to. They have subtly different semantics to arrays of unicode.

We should have some example client code for this before committing ourselves to a particular representation.

**Comment by hyanwong** [comment]
Sure. I have updated this so it should never return an object array anyway. But no hurry to get this in. It was to help @szhan and others who spent a fair bit of time comparing integer arrays between tree sequence before realising that they weren't actually comparable.

**Comment by hyanwong** [comment]
I think this also fixes #2181

**Review by benjeffery — APPROVED** [review]

**Pull request merged**
